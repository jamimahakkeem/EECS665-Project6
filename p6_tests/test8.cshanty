int fn(int n1, int n2){
int ans;
ans = n1+n2;
return ans;
}

void main()
{
report "hello";
fn(5,8);
<<<<<<< HEAD
}
=======
}
>>>>>>> jamima1
