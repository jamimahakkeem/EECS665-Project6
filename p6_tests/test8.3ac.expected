[BEGIN GLOBALS]
[str_0] "hello"
[END GLOBALS]
[BEGIN fn LOCALS]
n1 (formal arg of 8)
n2 (formal arg of 8)
ans (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
[END fn LOCALS]
fun_fn:     enter fn
            getarg 1 [n1]
            getarg 2 [n2]
            [tmp0] := [n1] ADD64 [n2]
            [ans] := [tmp0]
            setret [ans]
            goto lbl_0
lbl_0:      leave fn
[BEGIN main LOCALS]
tmp0 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            REPORT [[str_0]]
            setarg 1 5
            setarg 2 8
            call fn
lbl_1:      leave main
